lib/appsignal/hooks/puma.rb

```diff
@@ -33,5 +37,57 @@ def stop_workers
         end
       end
     end
+
+    class PumaProbe
+      def initialize
+        @hostname = Appsignal.config[:hostname] || Socket.gethostname
+      end
+
+      def call
+        return unless ::Puma.stats
+
+        stats = JSON.parse Puma.stats, :symbolize_names => true
+        counts = {
+          :backlog => 0,
+          :running => 0,
+          :pool_capacity => 0,
+          :max_threads => 0
+        }
+
+        if stats[:worker_status] # Multiple workers
+          stats[:worker_status].each do |worker|
+            stat = worker[:last_status]
+
+            counts[:backlog] += stat[:backlog]
+            counts[:running] += stat[:running]
+            counts[:pool_capacity] += stat[:pool_capacity]
+            counts[:max_threads] += stat[:max_threads]
+          end
+
+          gauge(:workers, stats[:workers], :kind => :count)
```

Comment:
The tag here is probably better named `type`?

Reply:
Ok! I cargo-culted this from the Elxir integration ( https://github.com/appsignal/appsignal-elixir/pull/462/files#diff-0acfe457d918e729ecd2d573f1fa9ef7R11 ) , if we want to keep a standard of sorts, we might have to rename them there before release too.

Reply:
yeah we should probably rename it there too. I think type is a better term than kind here